### PR TITLE
rusty-cpp: new recipe

### PR DESCRIPTION
**Pull request by seekstar**
### Summary
Changes to recipe:  **rusty-cpp**

#### Motivation

This is a library I find handy to use, and others may find it useful, too.

It is currently header-only. If I make it non-header-only in the future, should I modify the existing `conanfile.py`, or should I create a new folder and write a new `conanfile.py` there?

### CODE REVIEW

File: recipes/rusty-cpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+required_conan_version = ">=2.0"
+
+
+class RustyCppConan(ConanFile):
+    name = "rusty-cpp"
+    description = "Write C++ code like Rust!"
+    # dual licensed under the Apache License v2.0 and the MIT License
+    license = "Apache-2.0", "MIT"
```
Comment (valgur) [inline recipes/rusty-cpp/all/conanfile.py:15]:
```suggestion
    license = "Apache-2.0 OR MIT"
```

**Review by AbrilRBS — APPROVED** [review]
This looks good on my end, thanks a lot @seekstar for taking the time to submit this to CCI, we appreciate it.

Some notes:

> It is currently header-only. If I make it non-header-only in the future, should I modify the existing conanfile.py, or should I create a new folder and write a new conanfile.py there?

For this, the best approach would be to add a `header_only` option for the current recipe, (alongside the necessary compilation options) that toggles the header only behaviour as needed. See for example [`fmt`](https://github.com/conan-io/conan-center-index/blob/master/recipes/fmt/all/conanfile.py) that does something similar, but we can always help for your specific case when/if the times comes :)

- I've added a `package_id` method - header only libraries are the same for every platform, so this ensures only 1 package is needed everywhere
- I've added a min cppstd check for C++17 based on this line from [CMakeLists.txt](https://github.com/seekstar/rusty-cpp/blob/main/CMakeLists.txt#L5)

Otherwise, the recipe looks great!

File: recipes/rusty-cpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import os
+
+
+required_conan_version = ">=2.0"
+
+
+class RustyCppConan(ConanFile):
+    name = "rusty-cpp"
+    description = "Write C++ code like Rust!"
+    license = "Apache-2.0 OR MIT"
```
Comment (AbrilRBS) [inline recipes/rusty-cpp/all/conanfile.py:15]:
```suggestion
    license = "Apache-2.0", "MIT"
```

let's wait until Conan 2.x has better modelling of licenses to make this OR decision on behalf of the consumer, safer to have the list approach for now unless upstream clearly states its SPDX identifier

Comment (AbrilRBS) [reply recipes/rusty-cpp/all/conanfile.py:15]: Tracked as https://github.com/conan-io/conan/issues/17446

**Review by AbrilRBS — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Comment by seekstar** [comment]
Thank you very much. I've applied changes from this PR to the conanfile in the upstream repo.

The readme hasn't mentioned `conan` yet. I tried `conan install --requires="rusty-cpp/[*]` in a project using `rusty-cpp` but the command simply creates many shell scripts like `conanbuildenv-release-x86_64.sh`. I also checked [the document](https://docs.conan.io/2/reference/commands/install.html) but still don't know what's the correct usage of this command.

**Pull request merged**